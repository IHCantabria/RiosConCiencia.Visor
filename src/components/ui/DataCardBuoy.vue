<template>
  <div class="overviewcard">
    <div class="overviewcard__description">
      <vs-list class="additional-Info">
        <vs-list-item class="additional-Info__Icon">
          <i
            :class="[openLicense ? 'fas fa-unlock' : 'fas fa-lock']"
            :title="licenseType"
          ></i>
        </vs-list-item>
        <vs-list-item title="Latitude">
          <span>{{ latitude }}</span></vs-list-item
        >
        <vs-list-item title="Longitude"
          ><span>{{ longitude }}</span></vs-list-item
        >
      </vs-list>
      <vs-list class="variables-Info">
        <vs-list-header
          icon="info"
          title="Variables Info"
          color="primary"
        ></vs-list-header>
        <vs-list-item
          v-for="variable in variables"
          :key="variable.id"
          :title="variable.description"
        >
          <span><strong>Name</strong>: {{ variable.name }}</span>
          <span><strong>Type</strong>: {{ variable.type }}</span>
          <span><strong>Units</strong>: {{ variable.units }}</span>
          <span><strong>Minimum</strong>: {{ variable.minimum }}</span>
          <span><strong>Maximum</strong>: {{ variable.maximum }}</span>
        </vs-list-item>
      </vs-list>
    </div>
  </div>
</template>
<script>
export default {
  props: {
    openLicense: {
      type: Boolean,
      required: false,
      default: () => {
        return false;
      }
    },
    licenseType: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    longitude: {
      type: Number,
      required: false,
      default: () => {
        return 0;
      }
    },
    latitude: {
      type: Number,
      required: false,
      default: () => {
        return 0;
      }
    },
    variables: {
      type: Array,
      required: false,
      default: () => {
        return [];
      }
    }
  }
};
</script>
<style lang="scss" scoped>
.overviewcard {
  border-top: solid $color-primary;
  border-bottom: solid $color-primary;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  background-color: $color-very-light-grey;

  &__description {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
}
</style>
