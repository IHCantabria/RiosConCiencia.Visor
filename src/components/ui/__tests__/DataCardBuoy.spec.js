import { shallowMount, createLocalVue } from "@vue/test-utils";
import DataCardBuoy from "../DataCardBuoy.vue";
import Vuesax from "vuesax";
import merge from "lodash.merge";

const localVue = createLocalVue();
localVue.use(Vuesax);

function createWrapper(overrides) {
  const defaultMountingOptions = { localVue };
  return shallowMount(DataCardBuoy, merge(defaultMountingOptions, overrides));
}

describe("dataCardBuoy.vue", () => {
  //SNAPSHOT TESTS
  it("renders correctly default values", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly with full info", () => {
    expect.assertions(1);
    const wrapper = createWrapper({
      propsData: {
        openLicense: true,
        licenseType: "open",
        longitude: 10,
        latitude: 30,
        variables: [
          {
            id: 1,
            name: "name test 1",
            type: "type test 1",
            units: "m",
            minimum: 14,
            maximum: 56
          },
          {
            id: 2,
            name: "name test 2",
            type: "type test 2",
            units: "s",
            minimum: 45,
            maximum: 156
          }
        ]
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
