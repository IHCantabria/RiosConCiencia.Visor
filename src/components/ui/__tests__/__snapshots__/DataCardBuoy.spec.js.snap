// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataCardBuoy.vue renders correctly default values 1`] = `
<div
  class="overviewcard"
>
  <div
    class="overviewcard__description"
  >
    <vs-list-stub
      class="additional-Info"
    >
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <i
          class="fas fa-lock"
          title=""
        />
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Latitude"
      >
        <span>
          0
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Longitude"
      >
        <span>
          0
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
     
    <vs-list-stub
      class="variables-Info"
    >
      <vs-list-header-stub
        color="primary"
        icon="info"
        iconpack="material-icons"
        title="Variables Info"
      />
       
    </vs-list-stub>
  </div>
</div>
`;

exports[`dataCardBuoy.vue renders correctly with full info 1`] = `
<div
  class="overviewcard"
>
  <div
    class="overviewcard__description"
  >
    <vs-list-stub
      class="additional-Info"
    >
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <i
          class="fas fa-unlock"
          title="open"
        />
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Latitude"
      >
        <span>
          30
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Longitude"
      >
        <span>
          10
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
     
    <vs-list-stub
      class="variables-Info"
    >
      <vs-list-header-stub
        color="primary"
        icon="info"
        iconpack="material-icons"
        title="Variables Info"
      />
       
      <vs-list-item-stub
        iconpack="material-icons"
      >
        <span>
          <strong>
            Name
          </strong>
          : name test 1
        </span>
         
        <span>
          <strong>
            Type
          </strong>
          : type test 1
        </span>
         
        <span>
          <strong>
            Units
          </strong>
          : m
        </span>
         
        <span>
          <strong>
            Minimum
          </strong>
          : 14
        </span>
         
        <span>
          <strong>
            Maximum
          </strong>
          : 56
        </span>
      </vs-list-item-stub>
      <vs-list-item-stub
        iconpack="material-icons"
      >
        <span>
          <strong>
            Name
          </strong>
          : name test 2
        </span>
         
        <span>
          <strong>
            Type
          </strong>
          : type test 2
        </span>
         
        <span>
          <strong>
            Units
          </strong>
          : s
        </span>
         
        <span>
          <strong>
            Minimum
          </strong>
          : 45
        </span>
         
        <span>
          <strong>
            Maximum
          </strong>
          : 156
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
  </div>
</div>
`;
